src/libs/localization-api/registry/translation.registry.ts

```diff
@@ -0,0 +1,70 @@
+import { hasDefaultExport, loadExtension } from '@umbraco-cms/backoffice/extension-api';
+import { UmbBackofficeExtensionRegistry } from '@umbraco-cms/backoffice/extension-registry';
+import { ReplaySubject, Subscription } from '@umbraco-cms/backoffice/external/rxjs';
+
+export type UmbTranslationDictionary = Map<string, string>;
+
+export class UmbTranslationRegistry {
+	#extensionRegistry;
+	#innerDictionary = new ReplaySubject<UmbTranslationDictionary>(1);
+	#innerDictionaryValue: UmbTranslationDictionary = new Map();
+	#subscription?: Subscription;
+
+	constructor(umbExtensionRegistry: UmbBackofficeExtensionRegistry) {
+		this.#extensionRegistry = umbExtensionRegistry;
+	}
+
+	get translations() {
+		return this.#innerDictionary.asObservable();
+	}
+
+	register(userCulture: string, fallbackCulture = 'en-us') {
```

Comment:
Wouldn't it make more sense to use a neutral culture (without any country/region, like `en`) or even the invariant culture (.NET uses an [empty string as name](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.invariantculture?view=net-7.0#remarks) or [`iv` as ISO code](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.twoletterisolanguagename?view=net-7.0#remarks))?

Reply:
Yes, but we had some trouble with the cultures of the backend, where the "en.xml" file is actually "en-gb", but the default is "en-us". I am currently reworking this into listening on the "lang" attribute in the browser, so makes sense to support both with and without the "region" part of the culture code, but there is a discrepancy with the backend here and not sure how to easily fix that.

Reply:
Hmm, on the server-side we can use [`CultureInfo.Parent`](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.parent?view=net-7.0) to get the fallback language and go from e.g. `en-GB` to `en`, so you don't have to manually provide the fallback language when defining the translations. And if it can't find translations within this fallback chain, it can use the app's default culture that's specified in the [`NeutralResourcesLanguageAttribute`](https://learn.microsoft.com/en-us/dotnet/api/system.resources.neutralresourceslanguageattribute?view=net-7.0).

It doesn't seem like browsers have a similar way to get 'parent' cultures, although [`Intl.Locale`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale) splits the language, script and region, so you could do the following:
```js
function getParent(tag) {
  var locale = new Intl.Locale(tag);
  if (locale.region && locale.script) return locale.language + '-' + locale.script;
  if (locale.region || locale.script) return locale.language;
  return null;
}

getParent('en-GB'); // en
getParent('en'); // null

getParent('zh-Hans-CN'); // zh-Hans
getParent('zh-Hans'); // zh
getParent('zh'); // null
```

The client side could then also define a similar neutral/default culture (as a single configuration value), in case the preferred culture can't be retrieved from the authenticated user or [`navigator.language`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/language) (or the first supported language from [`navigator.languages`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages)).